src/inmanta/module.py

```diff
@@ -1682,10 +1754,67 @@ def __init__(self, project: Optional[Project], path: str):
         except InvalidMetadata as e:
             raise InvalidModuleException(f"The module found at {path} is not a valid V2 module") from e
 
+    @classmethod
+    def get_installed_module(cls, project: Project, module_name: str) -> Optional["ModuleV2"]:
+        fq_module_path = f"{const.PLUGINS_PACKAGE}.{module_name}"
```

Comment:
Style wise, I have some questions about this one.

The method name looks like it is a simple factory method. 
But it is in fact a quite powerful unload/reload method, that makes some quite intricate assumptions on the environment.  (e.g. it hooks in to the `PluginModuleFinder` singleton). So it may behave in unexpected ways, which is not good. 

In a sense, this method is very robust, in that it can handle more complicated scenarios. In a sense it is not very robust, as it is implicitly coupled to the loader.  Up to now, in most cases, we do the unloading in fixtures and don't allow it during normal execution. So this is a bit of a different approach. 

So this method caught me a bit by surprise and I don't know very well what to think about. (So I'm not opposed to it, just uncertain). 

What are the pros/cons of doing the unloading here?

I do think it should get some more documentation.

Reply:
The idea was to handle the scenario where a V1 module was already loaded, but in practice that can not happen. I don't think there is a reason to keep the unloading logic in there, so I will remove it.